app/controllers/concerns/locale.rb

```diff
@@ -0,0 +1,51 @@
+module Concerns::Locale
+  extend ActiveSupport::Concern
+
+  included do
+    before_filter :set_locale
+  end
+
+  def explicitly_requested_language
+    params[:locale]
+  end
+
+  def user_settings_language
+    current_user.locale if current_user
+  end
+
+  def session_language
+    session[:locale]
+  end
+
+  def browser_language
+    request.env['HTTP_ACCEPT_LANGUAGE'] ? request.env['HTTP_ACCEPT_LANGUAGE'].scan(/^[a-z]{2}/).first : nil
+  end
+
+  def default_language
+    FoodsoftConfig[:default_locale] || ::I18n.default_locale
```

Comment:
do we need the `::` prefix? We don't use at the other places...

Reply:
My guess would be: this concern can be included at places where there might e.g. be an `I18n` submodule defined, and in that case we'd still want to reference the global `I18n` module.